lib/rubygems/specification.rb

```diff
@@ -2711,11 +2711,18 @@ def validate packaging = true
         raise Gem::InvalidSpecificationException,
           "each license must be 64 characters or less"
       end
+
+      if !Gem::SPDX::IDENTIFIERS.include?(license) && !license.eql?(Gem::SPDX::NONSTANDARD)
+        warning <<-warning
+WARNING: license value '#{license}' is invalid.  Use a license identifier from
+http://spdx.org/licenses or '#{Gem::SPDX::NONSTANDARD}' for a nonstandard license.
+        warning
```

Comment:
The warning message now describes The Right Thing To Do for nonstandard licenses.